benchmarks/degredation_models.py

```diff
@@ -0,0 +1,549 @@
+import pybamm
+import numpy as np
+
+
+def compute_discretisation(model, param):
+    var_pts = {
+        pybamm.standard_spatial_vars.x_n: 20,
+        pybamm.standard_spatial_vars.x_s: 20,
+        pybamm.standard_spatial_vars.x_p: 20,
+        pybamm.standard_spatial_vars.r_n: 30,
+        pybamm.standard_spatial_vars.r_p: 30,
+        pybamm.standard_spatial_vars.y: 10,
+        pybamm.standard_spatial_vars.z: 10,
+    }
+    geometry = model.default_geometry
+    param.process_geometry(geometry)
+    mesh = pybamm.Mesh(geometry, model.default_submesh_types, var_pts)
+    return pybamm.Discretisation(mesh, model.default_spatial_methods)
+
+
+def solve_model_once(model, solver, t_eval):
+    solver.solve(model, t_eval=t_eval)
+
+
+class TimeBuildModelLossActiveMaterial:
+    param_names = ["model", "model option"]
+    params = (
+        [pybamm.lithium_ion.SPM, pybamm.lithium_ion.DFN],
+        ["none", "stress-driven", "reaction-driven", "stress and reaction-driven"],
+    )
+
+    def time_setup_model(self, model, params):
+        self.param = pybamm.ParameterValues("Ai2020")
+        self.model = model({"loss of active material": params})
+        self.param.process_model(self.model)
+        compute_discretisation(self.model, self.param).process_model(self.model)
+
+
+class TimeBuildSimulationLossActiveMaterial:
+    param_names = ["model", "with experiment", "model option"]
+    params = (
+        [pybamm.lithium_ion.SPM, pybamm.lithium_ion.DFN],
+        [False, True],
+        ["none", "stress-driven", "reaction-driven", "stress and reaction-driven"],
+    )
+
+    def time_setup_simulation(self, model, with_experiment, params):
+        self.param = pybamm.ParameterValues("Ai2020")
+        self.model = model({"loss of active material": params})
+        if with_experiment:
+            exp = pybamm.Experiment(
+                [
+                    "Discharge at 0.1C until 3.105 V",
+                ]
+            )
+            pybamm.Simulation(self.model, parameter_values=self.param, experiment=exp)
+        else:
+            pybamm.Simulation(self.model, parameter_values=self.param, C_rate=1)
+
+
+class TimeSolveLossActiveMaterial:
+    param_names = ["model", "solve first", "model option"]
+    params = (
+        [pybamm.lithium_ion.SPM, pybamm.lithium_ion.DFN],
+        [False, True],
+        ["none", "stress-driven", "reaction-driven", "stress and reaction-driven"],
+    )
+
+    solver = pybamm.CasadiSolver()
+
+    def setup(self, model, solve_first, params):
+        self.model = model({"loss of active material": params})
+        c_rate = 1
+        tmax = 4000 / c_rate
+        nb_points = 500
+        self.t_eval = np.linspace(0, tmax, nb_points)
+        geometry = self.model.default_geometry
+
+        # load parameter values and process model and geometry
+        param = pybamm.ParameterValues("Ai2020")
+        param.process_model(self.model)
+        param.process_geometry(geometry)
+
+        # set mesh
```

Comment:
Can these lines be replaced by `compute_discretisation`? (It appears multiple times below, so not repeating the comment)